src/cpp/cpp_constructor.cpp

```diff
@@ -223,9 +223,9 @@ optionalt<codet> cpp_typecheckt::cpp_constructor(
 
     side_effect_expr_function_callt function_call(
       cpp_namet(constructor_name, source_location).as_expr(),
-      operands_tc);
-
-    function_call.add_source_location()=source_location;
+      operands_tc,
+      typet(),
```

Comment:
Is the default constructor for typet() really supported / the best option?

Reply:
#3941 introduces `uninitialized_typet`, which might be a better option. I'll make this PR depend on #3941.